python/IAM_USER_PERMISSION_BOUNDARY_CHECK/IAM_USER_PERMISSION_BOUNDARY_CHECK.py

```diff
@@ -0,0 +1,489 @@
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+
+    ###############################
+    # Add your custom logic here. #
+    ###############################
+    iam_client = get_client('iam', event)
+    compliance_results = []
+    users_list = iam_client.list_users()
+    permission_boundary_list = iam_client.list_policies(OnlyAttached=True, PolicyUsageFilter='PermissionsBoundary')
+    if users_list and permission_boundary_list:
+        for user in users_list['Users']:
+            compliance_type = evaluate_user(event, user['UserName'], valid_rule_parameters)
+            compliance_results.append(build_evaluation(user['UserId'], compliance_type, event, "AWS::IAM::User"))
+        return compliance_results
+
+    return 'NOT_APPLICABLE'
+
+def evaluate_user(event, username, valid_rule_parameters):
+    iam_client = get_client('iam', event)
+    user_details = iam_client.get_user(UserName=username)
+    try:
+        if user_details['User']['PermissionsBoundary']:
+            try:
+                if valid_rule_parameters['policyArns']:
+                    permission_boundary_policy_names = valid_rule_parameters['policyArns'].replace(" ", "")
+                    permission_boundary_policy_name_list = permission_boundary_policy_names.split(",")
+                    boundary_name = user_details['User']['PermissionsBoundary']['PermissionsBoundaryArn']
+                    for permission_policy_name in permission_boundary_policy_name_list:
+                        if permission_policy_name == boundary_name:
+                            complaince_response = 'COMPLIANT'                  # the IAM user has the specific permission boundary attached
+                            break
+                        else:
+                            complaince_response = 'NON_COMPLIANT'              # the IAM user does not have the specific permission boundary attached
+            except:
+                complaince_response = 'COMPLIANT'                              # the IAM user has the permission boundary attached no specific input provided.
+    except:
+        complaince_response = 'NON_COMPLIANT'                                  # the IAM user has no permission boundary attached no specific input provided.
+
+    return complaince_response
+
+def evaluate_parameters(rule_parameters):
+    """Evaluate the rule parameters dictionary validity. Raise a ValueError for invalid parameters.
+
+    Return:
+    anything suitable for the evaluate_compliance()
+
+    Keyword arguments:
+    rule_parameters -- the Key/Value dictionary of the Config Rules parameters
+    """
+    valid_rule_parameters = rule_parameters

```

Comment:
No, works on rule_parameter, then assign the valid parameter only when it is valid.